nixos/modules/services/networking/networkmanager-connections/wifi.nix

```diff
@@ -0,0 +1,49 @@
+{ pkgs, lib, config, ... }: with lib; {
+  options = {
+    password = mkOption {
+      type = types.nullOr types.str;
+      description = "The WPA2 password of the connection";
```

Comment:
I'd make it very clear in the description that this will end up world-readable in the store